### DISCUSSION TITLE
Seemingly nonsensical Minder lag

### DISCUSSION THREAD

**Discussion by Sebast1986** [phase1geo/Minder#415] [Q&A]
_First of all, excuse me if I don't write well, English is not my natal language._ 

(Si alguien de acá habla castellano, español y se le dificulta el inglés puedo traducir todo este texto. Sólo díganme y lo hago)

**The problem:** 

Minder lag when I write inside nodes. The letters take time to appear or directly they do not, similar to when you use the predictive on an old cell phone.

 **What draws my attention:**

Although my computer is old, Minder is not saturating my resources, it hardly uses them. I tried to assign top priority to it but nothing changed.

**How I installed it:**

I did it manually following the instructions. I installed each dependency one by one (several I already had) and then compiling using the "sudo ./app install" instruction

**My system:** 

Linux Mint 20.2 with Mate desktop and nvidia drivers

I can film the problem and upload it for you to see, put screenshots of what you want or take any test that you tell me.

My knowledge is very basic, so I would really appreciate simple instructions as if you were explaining to your grandparents or an old man (like the ones you put in the installation instructions).

I have a real interest in solving it, I tested all the best known programs of this type (free and paid) and Minder is the only one with which I can work dynamically since it is simple, it is not saturated with unnecessary options and you can create maps simply writing as if it were a text.

Thank you very much for your time.

**Comment by phase1geo** [comment]
Could you attach (or email me if you would prefer) the saved minder file?

**Reply by Sebast1986** [reply]
Wow you really responded quickly! **Thank You.**

 Yes I can send the saved files and I can upload any folder that you need to google drive. 

I tried on different computers and the same thing happened, but all the computers have Linux Mint installed, 

Maybe the problem has to do with that? I read in the "issues" section that a user had a problem when starting the program, it started slowly ("Startup is very slow #244" by dagelf) and it was related to "Gnome dbus timeout" and Mate is derived from Gnome, 
maybe some gtk related problem? sorry if what i'm saying is nonsense i really don't understand much. Now I attach the saves. Thanks again.
The saves ---------> [https://drive.google.com/drive/folders/1I8XCsMNo_wvHDuMdVocIkaDzm4OrtT-a?usp=sharing](url)

**Comment by phase1geo** [comment]
I was able to load up both mind maps in the latest version of Minder for elementary OS (version 1.13.1).  I didn't have any problems with application lags there.

My general thoughts about the problem could be a few things:

1. Are you using version 1.13.1 of Minder on Mint (I'm not sure what the latest available version is on that distribution).  It would be most useful and beneficial for you to use the latest version.
2. Are you using a Debian package or a Flathub Flatpak image of Minder?  There may be problems with the Flatpak version due to the version of valac that it currently comes with.
3. You could always try building from source; however, you will need to make sure that your valac version is version 0.50 or below (ideally).
4. You could try to download and run the elementary Flatpak.  That should run clean and would be the ideal solution for you.  I don't know if this is possible yet or not though.  You could try to run the following commands in a Terminal:

flatpak remote-add --if-not-exists --system appcenter https://flatpak.elementary.io/repo.flatpakrepo
flatpak install -y appcenter io.elementary.Platform io.elementary.Sdk

I'm not a Flatpak expert, especially on OSes other than elementary, so I may not be able to get you the information that you need.

**Reply by Sebast1986** [reply]
With both the flatpak and the software manager I had a problem so I installed from the sources. 
I followed these instructions [https://github.com/phase1geo/Minder](url) and used this git [https://github.com/phase1geo/Minder.git](url)

Yes, I'm using **Minder 1.13.1** I'm not using the one from the repositories but the one from github. 

I'm using Valac 0.48.6 ("0.48.6-0ubuntu1" it is literally as it appears in the package manager, 
maybe some difference due to the ubuntu origin?)

Right now I'm going to test what you told me about flatpack. What I provided from flatpak was through the software manager. Linux Mint usually includes not only the original program from its repositories but also a flatpak alternative. Sometimes they are more up-to-date, sometimes older.

**Reply by Sebast1986** [reply]
The problem persists. I tried the commands you gave me, which I imagine are to install the ElementaryOs appcenter. The system allowed me to run and install them but not use them. When trying to add only the repository, it appeared to me as previously installed, I calculate that because Mint has built-in flatpak repositories.

The flatpak version and the linux Mint version, both show the software with window and icon colors, while the one that I installed from the source shows them gray (as I have them configured in the system, as if they managed the windows differently )
At the moment the software is working fine (approx 15 seconds) and at times it has lag, it is like an intermittence, similar to when the system is overloaded and the programs get stuck. As I said before, I tried on another pc, better than mine (which is very old) despite the fact that it also consumes almost no resources. What both pcs have in common is the Linux Mint system, and the same "Mate" desktop.

Thank you again for your time, and tell me any tests you want me to do so I don't steal so much time from you. Or if there is something to read that can guide me to find the answer without depending on you, etc.
